smac/smbo/smbo.py

```diff
@@ -126,12 +127,16 @@ def run(self):
 
             self.logger.debug("Intensify")
 
-            self.incumbent, inc_perf = self.intensifier.intensify(
-                challengers=challengers,
-                incumbent=self.incumbent,
-                run_history=self.runhistory,
-                aggregate_func=self.aggregate_func,
-                time_bound=max(0.01, time_spend))
+            try: self.incumbent, inc_perf = self.intensifier.intensify(
+                  challengers=challengers,
+                  incumbent=self.incumbent,
+                  run_history=self.runhistory,
+                  aggregate_func=self.aggregate_func,
+                  time_bound=max(0.01, time_spend))
+            except (TAEAbortException, BudgetExhaustedException):
```

Comment:
`intensify` races more than one challenger against the incumbent
and we always want to return the latest incumbent.
Here a potential problem:
If the first challenger beats the incumbent, and on the second challenge we would get a `BudgetExhaustedException`,
we would not return the newest incumbent (i.e., the first challenger) with your implementation.

Either these exceptions will be handled within intensify(),
or at least Intensifier stores the current incumbent always in an object variable,
such that we can return `self.intensifier.incumbent`